types/index.d.ts

```diff
@@ -0,0 +1,307 @@
+         * Defaults to header
+         */
+        authorizationMethod?: 'header' | 'body' | undefined;
+    } | undefined;
+}
+
+export type TokenType = "access_token" | "refresh_token";
+
+export interface Token {
+    [x: string]: unknown;
+}
+
+export interface AccessToken {
+    /**
+     * Immutable object containing the token object provided while constructing a new access token instance.
+     * This property will usually have the schema as specified by RFC6750,
+     * but the exact properties may vary between authorization servers.
+     */
+    token: Token;
+
+    /**
+     * Determines if the current access token is definitely expired or not
+     * @param expirationWindowSeconds Window of time before the actual expiration to refresh the token. Defaults to 0.
+     */
+    expired(expirationWindowSeconds?: number): boolean;
+
+    /**
+     * Refreshes the current access token
+     *
+     * @param params Optional argument for additional API request params.
+     * @param [params.scope] String or array of strings representing the application privileges
+     * @param [httpOptions] Optional http options passed through the underlying http library
+     */
+    refresh(params?: {
+        scope?: string | string[] | undefined;
+    }, httpOptions?: WreckHttpOptions): Promise<AccessToken>;
+
+    /**
+     * Revokes either the access or refresh token depending on the {tokenType} value
+     *
+     * @param tokenType A string containing the type of token to revoke (access_token or refresh_token)
+     * @param [httpOptions] Optional http options passed through the underlying http library
+     */
+    revoke(tokenType: TokenType, httpOptions?: WreckHttpOptions): Promise<void>;
+
+    /**
+     * Revokes both the current access and refresh tokens
+     *
+     * @param [httpOptions] Optional http options passed through the underlying http library
+     */
+    revokeAll(httpOptions?: WreckHttpOptions): Promise<void>;
+}
+
+export interface WreckHttpOptions {
+    baseUrl?: string | undefined;
+    socketPath?: string | undefined;
+    payload?: unknown;
+    headers?: { [key: string]: unknown } | undefined;
+    redirects?: false | number | undefined;
+    redirect303?: boolean | undefined;
+    beforeRedirect?: ((
+        redirectMethod: string,
+        statusCode: number,
+        location: string,
+        resHeaders: { [key: string]: unknown },
+        redirectOptions: unknown,
+        next: () => {},
+    ) => void) | undefined;
+    redirected?: ((statusCode: number, location: string, req: unknown) => void) | undefined;
+    timeout?: number | undefined;
+    maxBytes?: number | undefined;
+    rejectUnauthorized?: boolean | undefined;
+    agent?: unknown;
+    secureProtocol?: string | undefined;
+    ciphers?: string | undefined;
+    events?: boolean | undefined;
+    json?: boolean | "strict" | "force" | undefined;
+    gunzip?: boolean | "force" | undefined;
+}
```

Comment:
Replace with the types published by `@hapi/wreck`: https://github.com/hapijs/wreck/blob/f53808548f8253c194da0f32ac9ae3d986e86d79/lib/index.d.ts#L194